code/modules/mob/living/carbon/human/species_types.dm

```diff
@@ -519,3 +519,191 @@ var/global/image/plasmaman_on_fire = image("icon"='icons/mob/OnFire.dmi', "icon_
 		return 0
 	return 1
 
+
+
+
+var/global/list/synth_flesh_disguises = list()
+
+/datum/species/synth
+	name = "Synth" //inherited from the real species, for health scanners and things
+	id = "synth"
+	say_mod = "beep boops" //inherited from a user's real species
+	sexes = 0
+	specflags = list(NOTRANSSTING,NOBREATH,VIRUSIMMUNE) //all of these + whatever we inherit from the real species
+	safe_oxygen_min = 0
+	safe_toxins_min = 0
+	safe_toxins_max = 0
+	safe_co2_max = 0
+	SA_para_min = 0
+	SA_sleep_min = 0
+	dangerous_existence = 1
+	need_nutrition = 0 //beep boop robots do not need sustinance
+	meat = null
+	var/list/initial_specflags = list(NOTRANSSTING,NOBREATH,VIRUSIMMUNE) //for getting these values back for assume_disguise()
+	var/disguise_fail_health = 75 //When their health gets to this level their synthflesh partially falls off
+	var/image/damaged_synth_flesh = null //an image to display when we're below disguise_fail_health
+	var/datum/species/fake_species = null //a species to do most of our work for us, unless we're damaged
+
+
+/datum/species/synth/military
+	name = "Military Synth"
+	id = "military_synth"
+	armor = 25
+	punchmod = 10
+	disguise_fail_health = 50
+
+
+/datum/species/synth/admin_set_species(mob/living/carbon/human/H, old_species)
+	assume_disguise(old_species,H)
+
+
+/datum/species/synth/handle_chemicals(datum/reagent/chem, mob/living/carbon/human/H)
+	if(chem.id == "synthflesh")
```

Comment:
it's how it's all done I'm afraid, check every other handle_chemicals, and everything else about the reagent system, it's all chem.id checks.